lua/core/menu/devices.lua

```diff
@@ -130,25 +133,41 @@ m.enc = function(n,delta)
   end
 end
 
-local function redraw_keyboard_layout_select()
-  local len = tabutil.count(m.options["keyboard layout"])
+local function redraw_select(do_show_all_if_fit, do_uppercase)
   screen.clear()
-  for i=1,6 do
-    if (i > 2 - m.pos + 1) and (i < m.len - m.pos + 3 + 1) then
-      local name = m.options["keyboard layout"][i+m.pos-2 - 1]
-      local line = string.upper(name)
-      screen.level(i==3 and 15 or 4)
+
+  if do_show_all_if_fit==nil then do_show_all_if_fit = true end
+  if do_uppercase==nil then do_uppercase = false end
+  local len = tabutil.count(m.options[m.section])
+
+  if do_show_all_if_fit and len <= 6 then --  -> no scroll
+    for i=1,len do
+      local line = m.options[m.section][i]
+      if do_uppercase then line = string.upper(line) end
+      screen.level(i==m.pos and 15 or 4)
       screen.move(20,10*i)
       screen.text(line)
     end
+  else --  -> scroll & selected item is always at pos 3
+    for i=1,6 do
+      if (i > 2 - m.pos + 1) and (i < m.len - m.pos + 3 + 1) then
+        local line = m.options[m.section][i+m.pos-2 - 1]
+        if do_uppercase then line = string.upper(line) end
+        screen.level(i==3 and 15 or 4)
+        screen.move(20,10*i)
+        screen.text(line)
+      end
+    end
   end
+
   screen.update()
 end
 
 m.redraw = function()
-
-  if (m.section == "keyboard layout" and m.mode == "select") then
-    redraw_keyboard_layout_select()
+  if m.mode == "select" then
+    local do_show_all_if_fit = false
```

Comment:
i disabled the "show all if less than 6 items" (https://github.com/monome/norns/pull/1408#issuecomment-909635580) as the absence of top padding forces a cognitive shift.

(in my brain i associated the top padding w/ "oh, i'm in the select screen" and i'm kinda lost w/ its absence)

you can play w/ the value to see for yourself.